Tests/Honeycomb/HoneycombOptionsTests.swift

```diff
@@ -531,14 +547,91 @@ final class HoneycombOptionsTests: XCTestCase {
     func testServiceNameFromResourceAttributes() throws {
         let data = [
             "HONEYCOMB_API_KEY": "key",
-            "OTEL_RESOURCE_ATTRIBUTES": "service.name=better",
+            "OTEL_RESOURCE_ATTRIBUTES": "service.name=better,service.version=1",
         ]
         let source = HoneycombOptionsSource(info: data)
         let options = try HoneycombOptions.Builder(source: source).build()
 
         XCTAssertEqual("better", options.serviceName)
+        XCTAssertEqual("1", options.serviceVersion)
         let expectedResources = [
             "service.name": "better",
+            "service.version": "1",
+            "honeycomb.distro.version": honeycombLibraryVersion,
+            "honeycomb.distro.runtime_version": runtimeVersion,
+        ]
+        XCTAssertEqual(expectedResources, options.resourceAttributes)
+    }
+
+    func testServiceNameSetterTakesPrecedence() throws {
+        let data = [
+            "HONEYCOMB_API_KEY": "key",
+            "OTEL_SERVICE_NAME": "explicit",
+            "OTEL_SERVICE_VERSION": "1",
+        ]
+        let source = HoneycombOptionsSource(info: data)
+        let options = try HoneycombOptions.Builder(source: source)
+            .setServiceName("override")
+            .setServiceVersion("2")
+            .build()
+
+        XCTAssertEqual("override", options.serviceName)
+        XCTAssertEqual("2", options.serviceVersion)
+        let expectedResources = [
+            "service.name": "override",
+            "service.version": "2",
+            "honeycomb.distro.version": honeycombLibraryVersion,
+            "honeycomb.distro.runtime_version": runtimeVersion,
+        ]
+        XCTAssertEqual(expectedResources, options.resourceAttributes)
+    }
+
+    func testResourceAttributeSourceTakesPrecedenceOverSetters() throws {
+        let data = [
+            "HONEYCOMB_API_KEY": "key",
+            "OTEL_SERVICE_NAME": "explicit",
+            "OTEL_SERVICE_VERSION": "2",
+            "OTEL_RESOURCE_ATTRIBUTES": "service.name=resource,service.version=1",
+        ]
+        let source = HoneycombOptionsSource(info: data)
+        let options = try HoneycombOptions.Builder(source: source)
+            .setServiceName("override")
+            .setServiceVersion("3")
+            .build()
+
+        XCTAssertEqual("override", options.serviceName)
+        XCTAssertEqual("3", options.serviceVersion)
+        let expectedResources = [
+            "service.name": "resource",
+            "service.version": "1",
+            "honeycomb.distro.version": honeycombLibraryVersion,
+            "honeycomb.distro.runtime_version": runtimeVersion,
+        ]
+        XCTAssertEqual(expectedResources, options.resourceAttributes)
+    }
+
+    func testResourceAttributesTakePrecedenceOverSetters() throws {
```

Comment:
we should have the same discussion about these tests as on https://github.com/honeycombio/honeycomb-opentelemetry-android/pull/60#discussion_r1995671524